code/game/objects/items/weapons/dna_injector.dm

```diff
@@ -18,10 +18,10 @@
 	var/datatype = 0
 	var/value = 0
 
-/obj/item/dnainjector/New()
+/obj/item/dnainjector/proc/setbuf()

```

Comment:
Try setting this to `Intitialize` and not making any other changes other than converting the `New`'s to `Initialize`. This shouldn't fail...

Reply:
Done & retested. Still works.